library/src/main/java/com/google/android/exoplayer/text/tx3g/TextParser.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2014 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer.text.tx3g;
+
+import com.google.android.exoplayer.text.Subtitle;
+import com.google.android.exoplayer.text.SubtitleParser;
+import com.google.android.exoplayer.util.MimeTypes;
+
+import android.util.Log;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * A simple Text parser that supports Tx3g presentation profile.
+ * <p>
+ * Supported features in this parser are:
+ * <ul>
+ *   <li>content
+ *   <li>core
+ *   <li>presentation
+ *   <li>profile
+ *   <li>structure
+ *   <li>time-offset
+ *   <li>timing
+ *   <li>tickRate
+ *   <li>time-clock-with-frames
+ *   <li>time-clock
+ *   <li>time-offset-with-frames
+ *   <li>time-offset-with-ticks
+ * </ul>
+ * </p>
+ * @see <a href="http://www.w3.org/TR/ttaf1-dfxp/">TTML specification</a>
+ */
+public class TextParser implements SubtitleParser {
+  private static final String TAG = "TextParser";
+
+
+  private final List<SubtitleData> mSubtitleList;
+
+  /**
+   * Equivalent to {@code TtmlParser(true)}.
+   */
+  public TextParser() {
+    Log.i(TAG,"TextParser ");
+    mSubtitleList = new LinkedList<SubtitleData>();
+  }
+
+
+  @Override
+  public Subtitle parse(InputStream inputStream, String inputEncoding, long startTimeUs)
```

Comment:
Right, but each call to parse is supposed to return a Subtitle containing just the single cue parsed during that call, not the accumulated result of all calls to parse so far.

The current approach can end up adding an arbitrarily large number of cues to the object, and will add duplicates if the user ever seeks backwards.